framework/test/unit/modules/poa/genesis_block_test_data.ts

```diff
@@ -0,0 +1,198 @@
+				},
+				...validators.slice(1, validators.length),
+			],
+			snapshotSubstore: {
+				activeValidators,
+				threshold,
+			},
+		},
+	],
+	[
+		'Invalid validator name character',
+		{
+			validators: [
+				{
+					...validators[0],
+					name: '@@@__++',
+				},
+				...validators.slice(1, validators.length),
+			],
+			snapshotSubstore: {
+				activeValidators,
+				threshold,
+			},
+		},
+		'`name` property is invalid. Must contain only characters a-z0-9!@$&_.',
+	],
+	[
+		'Not unique validator name',
+		{
+			validators: [
+				{
+					...validators[0],
+					name: validators[1].name,
+				},
+				...validators.slice(1, validators.length),
+			],
+			snapshotSubstore: {
+				activeValidators,
+				threshold,
+			},
+		},
+		'`name` property of all entries in the validators must be distinct.',
+	],
+	[
+		'Not unique validator address',
+		{
+			validators: [
+				{
+					...validators[0],
+					address: validators[1].address,
+				},
+				...validators.slice(1, validators.length),
+			],
+			snapshotSubstore: {
+				activeValidators,
+				threshold,
+			},
+		},
+		'`address` property of all entries in validators must be distinct.',
+	],
+	[
+		'validator address is not ordered',
+		{
+			validators: validators.slice(0).sort((a, b) => b.address.compare(a.address)),
+			snapshotSubstore: {
+				activeValidators,
+				threshold,
+			},
+		},
+		'`validators` must be ordered lexicographically by address.',
+	],
+	[
+		'active validator address is not unique',
+		{
+			validators,
+			snapshotSubstore: {
+				activeValidators: [
+					{
+						...activeValidators[0],
+						address: activeValidators[1].address,
+					},
+					...activeValidators.slice(1, activeValidators.length),
+				],
+				threshold,
+			},
+		},
+		'`address` properties in `activeValidators` must be distinct.',
+	],
+	[
+		'active validator address is not ordered',
+		{
+			validators,
+			snapshotSubstore: {
+				activeValidators: activeValidators.slice(0).sort((a, b) => b.address.compare(a.address)),
+				threshold,
+			},
+		},
+		'`activeValidators` must be ordered lexicographically by address property.',
+	],
+	[
+		'active validator address is missing from validators array',
+		{
+			validators,
+			snapshotSubstore: {
+				activeValidators: [
+					{ ...activeValidators[0], address: utils.getRandomBytes(20) },
+					...activeValidators.slice(1, activeValidators.length),
+				].sort((a, b) => a.address.compare(b.address)),
+				threshold,
+			},
+		},
+		'`activeValidator` address is missing from validators array.',
+	],
+	[
+		'active validator weight must be positive integer',
+		{
+			validators,
+			snapshotSubstore: {
+				activeValidators: [
+					{ ...activeValidators[0], weight: BigInt(-1) },
+					...activeValidators.slice(1, activeValidators.length),
+				],
+				threshold,
+			},
+		},
+		// This is the error message returned from schema validation
+		'Invalid terminate index. Index is 22580 but terminateIndex is 22579',
```

Comment:
I guess, the reason for this error message is that `weight` is `uint64` in the schema, so cannot be negative. For a more reasonable test maybe you could change `weight` to 0 (however I don't have any problems with the current test).

Reply:
Yes that is the reason. I will leave it as is.